internal/backend/remote-state/s3/client.go

```diff
@@ -109,6 +110,29 @@ func (c *RemoteClient) get(ctx context.Context) (*remote.Payload, error) {
 	var output *s3.GetObjectOutput
 	var err error
 
+	ctx, _ = attachLoggerToContext(ctx)
+
+	inputHead := &s3.HeadObjectInput{
+		Bucket: &c.bucketName,
+		Key:    &c.path,
+	}
+
+	// Head works around some s3 compatible backends not handling missing GetObject requests correctly (ex: minio Get returns Missing Bucket)
+	_, err = c.s3Client.HeadObject(ctx, inputHead)
+	if err != nil {
+		var nb *types.NoSuchBucket
+		if errors.As(err, &nb) {
+			return nil, fmt.Errorf(errS3NoSuchBucket, err)
+		}
+
+		var nk *types.NotFound
+		if errors.As(err, &nk) {
+			return nil, nil
+		}
+
+		return nil, err
+	}
```

Comment:
I'm pretty happy with this approach of using HeadObject. But my experience with the s3 api is a bit lacking. 

@tomasmik @marcinwyszynski @mbialon what do you guys think here? Is this an acceptable approach?

Reply:
I initially did this to work around an issue with MinIO not handing GetObject's correctly.  I was going to remove it, but noticed that Terraform was updated to do the same as part of their S3 Download Manager work (probably for buffer size allocation).

I could easily back it out, and am wondering if it's worth keeping as MinIO has a number of other issues.

Reply:
>what do you guys think here? 

I have no strong feelings about this. If it doesn't break compatibility, I think this is fine.